### ISSUE TITLE
Confusing pen and fill behaviour for vector heads

### ISSUE THREAD

**Issue by adigitoleo** [GenericMappingTools/GMT.jl#753]
Vector head fill and pen behaviour is inconsistent depending on nesting in the `arrow` NamedTuple.
Ideally, I would suggest to make it work only inside `arrow`, and disallow `pen` and `fill` keywords outside that context.
This makes it logically consistent: all arrow _head_ properties are set inside `arrow`.
However, other functions accept `pen` and `fill` keywords also, so I'm not sure of the best solution.

```julia
using GMT

function example_data()
    Gz = gmt("grdmath -R-2/2/-2/2 -I0.1 X Y R2 NEG EXP X MUL =")
    Gdzdx = gmt("grdmath ? DDX", Gz)
    Gdzdy = gmt("grdmath ? DDY", Gz)
    return Gdzdx, Gdzdy
end

function test_arrow_fill()
    subplot(grid = "2x2", panels_size = 10, name = "test_arrow_fill.png")
    dx, dy = example_data()
    subplot(:set, panel = 1)
    grdvector(
        dx,
        dy,
        inc = 0.3,
        arrow = (; length = 0.4, shape = 0, stop = true, norm = 0.4, pen = :cyan, fill = :red),
        linewidth = 1,
    )
    subplot(:set, panel = 2)
    grdvector(
        dx,
        dy,
        inc = 0.3,
        arrow = (; length = 0.4, shape = 0, stop = true, norm = 0.4, fill = :red),
        pen = :cyan,
        linewidth = 1,
    )
    subplot(:set, panel = 3)
    grdvector(  # Produces the expected result.
        dx,
        dy,
        inc = 0.3,
        arrow = (; length = 0.4, shape = 0, stop = true, norm = 0.4, pen = :cyan),
        fill = :red,
        linewidth = 1,
    )
    subplot(:set, panel = 4)
    grdvector(
        dx,
        dy,
        inc = 0.3,
        arrow = (; length = 0.4, shape = 0, stop = true, norm = 0.4),
        fill = :red,
        pen = :cyan,
        linewidth = 1,
    )
    subplot(:end)
end

test_arrow_fill()
```
![test_arrow_fill](https://user-images.githubusercontent.com/34595875/141254790-277cc1d3-dcc4-4cb4-8bc5-daa0259fec42.png)

**Comment by joa-quim** [comment]
I see. Yes, confusing but we are obliged to use at least one of ``fill`` or ``pen`` outside ``arrow()`` otherwise GMT barfs with
```
grdvector [ERROR]: Must specify at least one of -G, -W, -C
```
Need to work out some trickery here.

**Comment by joa-quim** [comment]
Can you please try #754 ?
It turns out that ``grdvector -W`` seems to be mandatory (looks like a bug to me), and since **-Q** (``arrow()``) is optional I didn't drop the ``fill`` and ``pen`` options from the base ``grdvector`` options, but I made them *optional*. Now you can set the ``fill`` and ``pen`` only from inside ``arrow()``, or alternatively from *base* ``grdvector``.

**Comment by adigitoleo** [comment]
According to discussion in the linked PR, I've opened two related issues on the GMT tracker:

- https://github.com/GenericMappingTools/gmt/issues/5997

- https://github.com/GenericMappingTools/gmt/issues/5998

**Comment by joa-quim** [comment]
I opened a related (double) issue too https://github.com/GenericMappingTools/gmt/issues/5988, including the transparency.

**Comment by adigitoleo** [comment]
Ah, oops I missed that one when I was looking through the tracker... Nice that the transparency is fixed.

**Comment by adigitoleo** [comment]
OK I think this can be closed, the fixes have been released in GMT 6.3 upstream.
